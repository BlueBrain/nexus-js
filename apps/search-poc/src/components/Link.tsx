
import React, { MouseEvent } from 'react';
import propTypes from 'prop-types';

import { HrefBuilderParams } from '../context/link';


const Link: React.FunctionComponent<{
  params: HrefBuilderParams
}> = ({ params, children }, { onLinkClick, buildHref }) => {
  const handleClick = (event: MouseEvent) => {
    event.preventDefault();
    onLinkClick(params);
  }
  const href = buildHref ? buildHref(params) : '#';

  return <a
    onClick={handleClick}
    href={href}
  >
    {children}
  </a>;
};

Link.contextTypes = {
  onLinkClick: propTypes.func,
  buildHref: propTypes.func,
}


export default Link;
