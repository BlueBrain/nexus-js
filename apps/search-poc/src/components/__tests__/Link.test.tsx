
import React from "react";

import Link from '../Link';
import { shallow } from "enzyme";


describe('Link component', () => {
  it("Renders correctly with props", () => {
    const props = {
      params: { selfUrl: '123', type: 'resource' },
    };

    const contextProps = {
      buildHref: jest.fn(),
      onResourceClick: jest.fn()
    };

    const result = shallow(<Link {...props}>Link</Link>, { context: contextProps });

    expect(result).toMatchSnapshot();
  });

  it('Renders # as href if buildHref is not provided', () => {
    const props = {
      params: { selfUrl: '123', type: 'resource' },
    }

    const contextProps = {
      onResourceClick: jest.fn(),
    };

    const linkNode = shallow(<Link {...props}>Link</Link>, { context: contextProps });
    expect(linkNode.find('a').prop('href')).toBe('#');
  });

  it('Renders correct href if buildHref is provided', () => {
    const props = {
      params: { selfUrl: '123', type: 'resource' },
    }

    const buildHref = params => `${params.type}/${params.selfUrl}`

    const contextProps = {
      buildHref,
      onResourceClick: jest.fn(),
    };

    const linkNode = shallow(<Link {...props}>Link</Link>, { context: contextProps });
    expect(linkNode.find('a').prop('href')).toBe(buildHref(props.params));
  });

  it('Calls onResourceClick function on click', () => {
    const props = {
      params: { selfUrl: '123', type: 'resource' },
    };

    const contextProps = {
      onResourceClick: jest.fn(),
    };

    const linkNode = shallow(<Link {...props}>Link</Link>, { context: contextProps });
    linkNode.find('a').simulate('click');

    expect(contextProps.onResourceClick.mock.calls.length).toBe(1);
    expect(contextProps.onResourceClick.mock.calls[0][0]).toBe(props.params);
  });
});
