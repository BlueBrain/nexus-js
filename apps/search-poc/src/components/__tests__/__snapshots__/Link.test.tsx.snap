// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component Renders correctly with props 1`] = `
<a
  onClick={[Function]}
>
  Link
</a>
`;
